homework-4/script.js

```diff
@@ -0,0 +1,247 @@
+      if (element.currentTarget.checked) { 
+        deleteButton.removeAttribute("disabled", false);
+      }   
+    });
+
+    editButton.addEventListener('click', (ele) => {
+      const modalEditObject = document.querySelector('.modal-edit-object');
+      modalEditObject.style.display = 'block';
+      let targetElement = ele.currentTarget;
+      let id = targetElement.getAttribute('data-id');
+
+      contactName.value = targetElement.getAttribute('data-name');
+      companyName.value = targetElement.getAttribute('data-company');
+      address.value = targetElement.getAttribute('data-address');
+      city.value = targetElement.getAttribute('data-city');
+      country.value = targetElement.getAttribute('data-country');
+
+      updateUsers.addEventListener('click', () => {
+        editElement(id);    
+      });              
+  });
+
+    tr.appendChild(checkBox);
+    tr.appendChild(tdCompanyName);
+    tr.appendChild(tdContactName);
+    tr.appendChild(tdAddress);
+    tr.appendChild(tdCity);
+    tr.appendChild(tdCountry);
+    tr.appendChild(editButton);
+    tr.appendChild(deleteButton);
+    tableBody.appendChild(tr);
+
+    tdCompanyName.innerText = element.name;
+    tdContactName.innerText = element.company;
+    tdAddress. innerText = element.address;
+    tdCity.innerText = element.city;
+    tdCountry.innerText = element.country;
+    deleteButton.innerHTML = '<i class="fa fa-close icon-close">';
+    editButton.innerHTML = '<i class="fa fa-pen icon-pen">';
+    editButton.className = 'edit';
+    deleteButton.className ='delete';
+    tdCompanyName.className ='name';
+    tr.className = 'add-tr';
+
+    // filter
+    function filterFunction(filterName, searchText) {
+      let tdName = tr.querySelectorAll('.' + filterName); 
+      for (i = 0; i < tdName.length; i++) {                
+        if (tdName[i].innerHTML.toUpperCase().indexOf(searchText.toUpperCase()) > -1) {
+          console.log(tdName);
+          tdName[i].style.display = '';
+        } else {
+          tdName[i].closest('tr').style.display = 'none';
+        }
+      }
+    }
+
+    const input = document.querySelector('.filter-button');
+    const inputFilters = document.querySelectorAll('.input');
+    input.addEventListener('click', function () {
+      let isFiltersFillded = false;
+      for (i = 0; i < inputFilters.length; i++) {
+        let val = inputFilters[i].value;
+        if (val.trim() != '') {
+          isFiltersFillded = true;  
+          let filterName = inputFilters[i].getAttribute('data-filter');
+          filterFunction(filterName, val.trim());
+        }
+      }
+      
+      if (!isFiltersFillded) {
+        showAll();
+      }
+    });
+
+    // Show all rows.
+    function showAll() {
+      let trs = document.querySelectorAll('tr.add-tr');
+      for (i = 0; i < trs.length; i++) {
+        trs[i].style.display = '';
+      }
+    }
+  }
+
+  // Edit element
+  function editElement(id) {
```

Comment:
to edit element you need not only id, but new data.

Please do not use data outside of function - pass it as argument.